File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.java
class Solution {
    public String getSmallestString(int n, int k) {
        StringBuilder ans=new StringBuilder();
        while(k>0){
            if((k - (n-1)*26) <=0){
                ans.append('a');
                k--;
            }else{
                char c=(char)(k - (n-1)*26 + 'a' - 1);
                ans.append(c);
                k-=(k - (n-1)*26);
            }
            n--;
        }
        return ans.toString();
    }
}

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
     string res = "";
        int i = n;
        while(k>0){
            int remain = k - (i-1);
            
            if(remain > 26){
                res += 'z';
                k = k-26;
            }
            
            else{
                k = k - remain;
                res += ('a' + remain - 1);
               
            }
            
            i--;
        }
        
        reverse(res.begin(), res.end());
        return res;   
    }
};